config/projects/validate.rb

```diff
@@ -0,0 +1,19 @@
+name "validate"
+
+package_name    "validate"
+install_dir     "/opt/validate"
+build_version   "0.0.1dev"
+build_iteration 1
+
+dependency 'preparation'
+
+Dir.glob('config/software/**').each do |filepath|
+  dep_name = ''
+  File.readlines(filepath).each do |line|
```

Comment:
Do we have any software definitions where the name of the software isn't the name of the file as well?  If not, we might be able to simplify this with something like: `dependency ::File.basename(filepath, ".rb")`.